scripts/gdb_printers.py

```diff
@@ -59,5 +61,259 @@ def build_pretty_printer():
+            return
+
+GeojsonPrinter()
+
+
+
+class SVGPrinter (gdb.Command):
+    """Display features on geojson.io."""
+
+    def __init__ (self):
+        super (SVGPrinter, self).__init__ ('svg', gdb.COMMAND_USER)
+        self.re_bbox = None
+        self.to_svg = {
+            'const osrm::engine::datafacade::ContiguousInternalMemoryDataFacade<osrm::engine::routing_algorithms::ch::Algorithm> &': self.Facade,
+            'const osrm::engine::datafacade::ContiguousInternalMemoryDataFacade<osrm::engine::routing_algorithms::corech::Algorithm> &': self.Facade,
+            'const osrm::engine::datafacade::ContiguousInternalMemoryDataFacade<osrm::engine::routing_algorithms::mld::Algorithm> &': self.Facade}
+
+
+    @staticmethod
+    def show_svg(svg, width, height):
+        svg = """<!DOCTYPE HTML>
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
+<head><meta http-equiv="content-type" content="application/xhtml+xml; charset=utf-8"/>
+  <style>
+    svg { background-color: beige; }
+    .node { stroke: #000; stroke-width: 4; fill: none; marker-end: url(#forward) }
+    .node.forward { stroke-width: 2; stroke: #0c0; font-family: sans; font-size: 42px }
+    .node.reverse { stroke-width: 2; stroke: #f00; font-family: sans; font-size: 42px }
+    .segment { marker-start: url(#osm-node); marker-end: url(#osm-node); }
+    .segment.weight { font-family: sans; font-size:24px; text-anchor:middle; stroke-width: 1;  }
+    .segment.weight.forward { stroke: #0c0; fill: #0c0; }
+    .segment.weight.reverse { stroke: #f00; fill: #f00; }
+    .edge { stroke: #00f; stroke-width: 2; fill: none; }
+    .edge.forward { stroke: #0c0; stroke-width: 1; marker-end: url(#forward) }
+    .edge.backward { stroke: #f00; stroke-width: 1; marker-start: url(#reverse) }
+    .edge.both { stroke: #fc0; stroke-width: 1; marker-end: url(#forward); marker-start: url(#reverse) }
+    .coordinates { font-size: 12px; fill: #333 }
+  </style>
+</head>
+<svg viewBox="0 0 """ + str(width) + ' ' + str(height) + """"
+     xmlns="http://www.w3.org/2000/svg"
+     xmlns:xlink="http://www.w3.org/1999/xlink">
+  <defs>
+    <marker id="forward" markerWidth="10" markerHeight="10" refX="9" refY="3" orient="auto" markerUnits="strokeWidth">
+      <path d="M0,0 C3,3 3,3 0,6 L9,3 z" fill="#000" />
+    </marker>
+
+    <marker id="reverse" markerWidth="10" markerHeight="10" refX="0" refY="3" orient="auto" markerUnits="strokeWidth">
+      <path d="M9,0 C6,3 6,3 9,6 L0,3 z" fill="#000" />
+    </marker>
+
+    <marker id="osm-node" markerWidth="10" markerHeight="10" refX="5" refY="5" orient="auto" markerUnits="strokeWidth">
+      <circle cx="5" cy="5" r="5"  fill="#000" />
+    </marker>
+  </defs>
+""" + svg + '\n</svg></html>'
+        fd, name = tempfile.mkstemp('.html')
+        os.write(fd, svg.encode('utf-8'))
+        os.close(fd)
+        print ('Saved to ' + name)
+        webbrowser.open('file://' + name)
+
+    @staticmethod
+    def Facade(facade, width, height, bbox):
```

Comment:
Do you think we could maybe improve readability by splitting this into separate functions?